app/scripts/directives/blockRegistration.js

```diff
@@ -24,6 +24,36 @@ angular.module('confRegistrationWebApp')
           if ($scope.block.endDateBlockId !== null) {
             DateRangeService.subscribe($scope.block.endDateBlockId, endDateChangeCallback);
           }
+
+          $scope.$watch('answer', function (answer, oldAnswer) {
+            if (angular.isUndefined(answer) || angular.isUndefined(oldAnswer) || angular.equals(answer, oldAnswer)) {
+              return;
+            }
+
+            $rootScope.$broadcast('answerChanged');
+
+            RegistrationCache.updateCurrent($scope.conference.id, $scope.currentRegistration);
+          }, true);
+
+          $scope.$on('answerChanged', function() {
+            if (!_.includes(['radioQuestion', 'selectQuestion', 'checkboxQuestion'], $scope.block.type)) {
```

Comment:
How about never registering the listener if the block is one of these types? Basically swap these 2 lines. That should make this only expensive if there are lots of multiple choice questions.